unique-universes/src/views/shoot.py

```diff
@@ -0,0 +1,295 @@
+    from src.database import PlayerData
+
+
+# acceleration directed to the center of Earth
+# measured in m/s^2
+EARTH_ACCELERATION = 9.81
+# same thing, just for the moon
+MOON_ACCELERATION = 1.62
+
+
+class AngleModal(disnake.ui.Modal):
+    def __init__(self, view: ShootMenu) -> None:
+        self.view = view
+
+        super().__init__(
+            title="Angle Input",
+            components=[
+                disnake.ui.TextInput(
+                    label="Angle degrees",
+                    custom_id="angle_deg",
+                    style=disnake.TextInputStyle.short,
+                    placeholder="15",
+                    value=str(self.view.stats.angle),
+                    min_length=1,
+                    max_length=3,
+                ),
+            ],
+            timeout=180,
+        )
+
+    async def on_timeout(self) -> None:
+        await self.view.on_timeout()
+        self.view.stop()
+
+    async def callback(self, interaction: disnake.ModalInteraction) -> None:
+        angle = interaction.text_values["angle_deg"]
+        if not angle.isdigit():
+            return await interaction.send("Invalid input! You can type only numbers (e.g 10, 45 ...)", ephemeral=True)
+
+        self.view.stats.angle = int(angle)
+        await interaction.send(f"Your new angle is {angle}", ephemeral=True)
+        return await self.view.update_message()
+
+
+@dataclasses.dataclass
+class ShootStats:
+    position: int = 0
+    angle: int = 15
+    ammunition: int = 5  # shots left
+    energy: int = 10  # angles adjustments left
+    ship_speed: int = 0
+    bullet_velocity: int = 0
+    bullet_type: str = "?"
+    obstacles_in_range: int = 8
+    enemy_position: int = 10
+    enemy_health: int = 10
+    enemy_energy: int = 5
+    enemy_has_vip_pass: bool = False
+    enemy_logins_in_a_row: int = 3
+    total_shots: int = 0
+    hits: int = 0
+    g_acc: float = dataclasses.field(
+        default_factory=lambda: random.uniform(  # noqa: S311
+            MOON_ACCELERATION,
+            EARTH_ACCELERATION,
+        ),
+    )
+
+    @property
+    def misses(self) -> int:
+        return self.total_shots - self.hits
+
+    @property
+    def angle_as_radians(self) -> float:
+        return math.radians(self.angle)
+
+    @property
+    def get_enemy_distance(self) -> int:
+        return abs(self.enemy_position - self.position)
+
+    def calculate_shot_range(self) -> float:
+        return (
+            2 * self.bullet_velocity * math.cos(self.angle_as_radians) * math.sin(self.angle_as_radians)
+        ) / self.g_acc
+
+    def calculate_max_possible_range(self) -> float:
+        return (self.bullet_velocity**2) / self.g_acc
+
+    def calculate_flight_time(self) -> float:
+        return (2 * self.bullet_velocity * math.sin(self.angle_as_radians)) / self.g_acc
+
+    def calculate_max_height(self) -> float:
+        return ((self.bullet_velocity**2) * (math.sin(self.angle_as_radians) ** 2)) / (2 * self.g_acc)
```

Comment:
How did you decide what should be a property and what shouldn't? Could all be properties if you removed `calculate` from the names. If the `calculate` methods were meant to be helper methods for other calculations, you could indicate that by starting their names with a `_`.